components/lib/dialog/Dialog.js

```diff
@@ -46,7 +47,9 @@ export const Dialog = React.forwardRef((inProps, ref) => {
     });
 
     useHandleStyle(DialogBase.css.styles, isUnstyled, { name: 'dialog' });
-
+    useOnEscape(maskRef, () => {
+        props.onHide();
+    });
```

Comment:
Dialog has the `pop.closeOnEscape` so I think your need to pass a `when` to your hook and pass `props.closeOnEscape` and also look for where its already being used in Dialog.js and not double escape it?

Reply:
@melloware Resolved and pushed the new changes. Thanks